src/solver/gemm.cpp

```diff
@@ -863,6 +863,13 @@ bool GemmFwdRest::IsApplicable(const ExecutionContext& context,
                                const ProblemDescription& problem) const
 {
 #if MIOPEN_USE_GEMM
+    const size_t two_gb_memory = 2ULL * 1024 * 1024 * 1024;
+    if((problem.GetInSize() >= two_gb_memory) || (problem.GetWeightsSize() >= two_gb_memory) ||
+       (problem.GetOutSize() >= two_gb_memory))
+    {
+        return false;
+    }
+
```

Comment:
Please add this to the base class instead of the derived classes. I see WRW missing from this PR as well

Reply:
only `GemmFwdRest`, `GemmWrwUniversal` and `GemmBwdRest` use the `int` limited offset [`Im2ColGPU`](https://github.com/ROCm/MIOpen/blob/develop/src/ocl/utilocl.cpp#L43) and [`Col2ImGPU`](https://github.com/ROCm/MIOpen/blob/develop/src/ocl/utilocl.cpp#L43). So, I avoided placing the check directly in base class, since it might restrict other solvers.